src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -285,11 +286,30 @@ class PythonFaaSCompletePlatforms(PexCompletePlatformsField):
     )
 
 
+class FaaSArchitecture(str, Enum):
+    X86_64 = "x86_64"
+    ARM64 = "arm64"
+
+
+class FaaSArchitectureField(StringField):
+    alias = "architecture"
+    valid_choices = FaaSArchitecture
+    expected_type = str
+    default = FaaSArchitecture.X86_64.value
+    help = help_text(
+        """
+        The architecture of the AWS Lambda runtime to target (x86_64 or arm64).
+        See https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html.
+        """
+    )
```

Comment:
What do you think about leaving this _field_ in the AWS lambda backend (given the docs are specific to AWS), and just have the enum be shared.

It's fine for the various request types to use `architecture: Optional[FaaSArchitecture] = None`, I think. (For this type, its value is all that's interesting, the field extras aren't used.)

Reply:
Done!